pkg/operator/ceph/pool/controller.go

```diff
@@ -422,23 +422,33 @@ func configureRBDStats(clusterContext *clusterd.Context, clusterInfo *cephclient
 	logger.Debug("configuring RBD per-image IO statistics collection")
 	namespaceListOpt := client.InNamespace(clusterInfo.Namespace)
 	cephBlockPoolList := &cephv1.CephBlockPoolList{}
-	var enableStatsForPools []string
+	var enableStatsForCephBlockPools []string
 	err := clusterContext.Client.List(clusterInfo.Context, cephBlockPoolList, namespaceListOpt)
 	if err != nil {
 		return errors.Wrap(err, "failed to retrieve list of CephBlockPool")
 	}
 	for _, cephBlockPool := range cephBlockPoolList.Items {
 		if cephBlockPool.GetDeletionTimestamp() == nil && cephBlockPool.Spec.EnableRBDStats {
 			// add to list of CephBlockPool with enableRBDStats set to true and not marked for deletion
-			enableStatsForPools = append(enableStatsForPools, cephBlockPool.ToNamedPoolSpec().Name)
+			enableStatsForCephBlockPools = append(enableStatsForCephBlockPools, cephBlockPool.ToNamedPoolSpec().Name)
 		}
 	}
-	logger.Debugf("RBD per-image IO statistics will be collected for pools: %v", enableStatsForPools)
 	monStore := config.GetMonStore(clusterContext, clusterInfo)
+	// Check for existing rbd stats pools
+	existingRBDStatsPools, e := monStore.Get("mgr", "mgr/prometheus/rbd_stats_pools")
+	if e != nil {
+		return errors.Wrapf(e, "failed to get rbd_stats_pools")
+	}
+
+	enableStatsForPools := strings.Join(enableStatsForCephBlockPools, ",") + existingRBDStatsPools
```

Comment:
The challenge with this approach is that `configureRBDStats()` is called every time there is a pool reconcile, which also happens again every time the operator is restarted, or an existing pool is edited. We can't keep appending the same pool names, so we will need a smarter merge. What about the following?
- Filter out the pools to append if they are already in the list
- When a pool is being deleted, make sure that pool is deleted from the list. `configureRBDStats()` is called in two places, when a pool is either created or deleted. When it is deleted, pass the name of the pool that is being deleted to `configureRBDStats()` (and it will be blank called called during pool creation). 

We will also need unit tests to check these scenarios.

Reply:
> The challenge with this approach is that `configureRBDStats()` is called every time there is a pool reconcile, which also happens again every time the operator is restarted, or an existing pool is edited. We can't keep appending the same pool names, so we will need a smarter merge. What about the following?
> 
I think we need to append same pool names every time bcz on config set it overrides the existing list of pool names, so I think the only way to resolve this is to check config for existing rbd stat pool names and set config with that list of names.
>     * Filter out the pools to append if they are already in the list
> 
>     * When a pool is being deleted, make sure that pool is deleted from the list. `configureRBDStats()` is called in two places, when a pool is either created or deleted. When it is deleted, pass the name of the pool that is being deleted to `configureRBDStats()` (and it will be blank called called during pool creation).
> 
Maybe that's already taken care of?
 https://github.com/rook/rook/blob/master/pkg/operator/ceph/pool/controller.go#L247 & 
https://github.com/rook/rook/blob/master/pkg/operator/ceph/pool/controller.go#L293
> 
> We will also need unit tests to check these scenarios.